portal/api-tests/login/login-with-sign-in-link.api-test.js

```diff
@@ -0,0 +1,103 @@
+jest.mock('csurf', () => () => (req, res, next) => next());
+jest.mock('../../server/routes/middleware/csrf', () => ({
+  ...(jest.requireActual('../../server/routes/middleware/csrf')),
+  csrfToken: () => (req, res, next) => next(),
+}));
+jest.mock('../../server/api', () => ({
+  login: jest.fn(),
+  sendSignInLink: jest.fn(),
+  loginWithSignInLink: jest.fn(),
+  validateToken: () => true,
+}));
+
+const { when } = require('jest-when');
+const app = require('../../server/createApp');
+const { get } = require('../create-api').createApi(app);
+const api = require('../../server/api');
+
+describe('GET /login/sign-in-link?t={signInToken}&u={userId}', () => {
+  const validSignInToken = '0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef';
+  const validUserId = '65626dc0bda51f77a78b86ae';
+  const userToken = 'a token';
+  const loginStatus = 'Valid username and password';
```

Comment:
Do we want to get this `loginStatus` from the constant or was this done as an intentional 'double-entry' check?

Reply:
Good idea, I'll do that